sqlmesh/core/snapshot/evaluator.py

```diff
@@ -90,7 +89,7 @@ def evaluate(
         start: TimeLike,
         end: TimeLike,
         execution_time: TimeLike,
-        snapshots: t.Dict[str, Snapshot],
+        snapshots: t.Collection[Snapshot],
```

Comment:
why did you change so much stuff from dict to collection?

Reply:
In most places when we deal with Snapshots we just gets the snapshots themselves. So rather than keying it on some opinionated key (with snapshot fqn or SnapshotId) we just decided to pass around a collection instead.